network/retrieval/retrieve.go

```diff
@@ -409,15 +419,18 @@ FINDPEER:
 	}
 
 	ret := &RetrieveRequest{
+		Ruid: uint(rand.Uint32()),
 		Addr: req.Addr,
 	}
-	protoPeer.logger.Trace("sending retrieve request", "ref", ret.Addr, "origin", localID)
+	protoPeer.logger.Trace("sending retrieve request", "ref", ret.Addr, "origin", localID, "ruid", ret.Ruid)
 	err = protoPeer.Send(ctx, ret)
 	if err != nil {
-		protoPeer.logger.Error("error sending retrieve request to peer", "err", err)
+		protoPeer.logger.Error("error sending retrieve request to peer", "ruid", ret.Ruid, "err", err)
```

Comment:
are you ok with logging and returning the error here?

Reply:
well, if i don't log the error here i lose the context of the peer address which is appended to the log line (since we are using the peer logger here)

Reply:
I know that I am a bit late, but we should use error annotations instead both logging them and returning to the caller.